go-controller/pkg/ovn/obj_retry.go

```diff
@@ -1016,22 +999,22 @@ func (oc *Controller) processObjectInTerminalState(objectsToRetry *retryObjs, ob
 // periodically or when explicitly requested.
 // Note: when applying WatchResource to a new resource type, the appropriate resource-specific logic must be added to the
 // the different methods it calls.
-func (oc *Controller) WatchResource(objectsToRetry *retryObjs) *factory.Handler {
+func (oc *Controller) WatchResource(objectsToRetry *retryObjs) (*factory.Handler, error) {
 	addHandlerFunc, err := oc.watchFactory.GetResourceHandlerFunc(objectsToRetry.oType)
 	if err != nil {
 		klog.Errorf("No resource handler function found for resource %v. "+
```

Comment:
this should be part of the returned error, then let the caller log it

Reply:
ack.